### PR TITLE
tests: Add main and filecheck tests for Toy

### PR DESCRIPTION
**Pull request by superlopuh**
I'm having trouble getting filecheck to cooperate. It refuses to run tests outside `tests/filecheck`, as far as I can tell, do you know what might be causing this?

```
(venv) (base) sasha@SL-MBP-16 ~/D/x/xdsl> lit -v docs/Toy/examples/ast.toy                                                           sasha/toy/filecheck!
lit: /Users/sasha/Developer/xdslproject/xdsl/venv/lib/python3.11/site-packages/lit/discovery.py:132: warning: unable to find test suite for 'docs/Toy/examples/ast.toy'
lit: /Users/sasha/Developer/xdslproject/xdsl/venv/lib/python3.11/site-packages/lit/discovery.py:281: warning: input 'docs/Toy/examples/ast.toy' contained no tests
error: did not discover any tests for provided path(s)
```

Running the copy in `filecheck/frontend` works fine

### CODE REVIEW

**Comment by webmiche** [comment]
Did you add the proper lit.cfg files?

**Comment by superlopuh** [comment]
It might not be the end of the world to duplicate the files, and have the tests just in the filecheck... And we could also point to the filecheck folder in the tutorial, it just seems less nice than running filecheck on `docs/Toy/toy/tests/filecheck`

**Comment by webmiche** [comment]
Here the [docu](https://llvm.org/docs/CommandGuide/lit.html#test-discovery).

What is it you are trying to achieve? Run the toy tests as part of our already existing lit test suite? Or run them in isolation?

For the former, you probably need to move the tests to the preexisting folder. For the latter, copying/extending the lif.cfg would probably be the way to go. We can also just add a second lit invocation to the CI, if that is what you are tryign to achieve :sweat_smile:

File: docs/Toy/examples/ast.toy [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# RUN: toyc-ch1 %s -emit=ast 2>&1 | FileCheck %s
```
Comment (superlopuh) [inline docs/Toy/examples/ast.toy:1]: This test passes as-is exactly from the LLVM repo :)

**Review by webmiche — APPROVED** [review]
LGTM now :+1:

**Review by math-fehr — APPROVED** [review]
:+1:

**Pull request merged**